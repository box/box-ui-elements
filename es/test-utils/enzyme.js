function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == typeof i ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != typeof t || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != typeof i) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
/* eslint-disable import/no-extraneous-dependencies */
import * as React from 'react';
import { mount as baseMount, shallow as baseShallow } from 'enzyme';

// Global providers that need setup
import { ThemeProvider } from 'styled-components';
import defaultTheme from '../styles/theme';

// Wrap this around every component so they can use app-wide context.
// Prevents breaking due to missing providers.
// See https://github.com/airbnb/enzyme/blob/master/docs/api/ReactWrapper/getWrappingComponent.md
/* eslint-disable react/prop-types */
const Wrappers = ({
  children
}) => {
  return /*#__PURE__*/React.createElement(ThemeProvider, {
    theme: defaultTheme
  }, children);
};

/**
 * mount() from Enzyme but with required app-wide context providers included by default
 */
const mountConnected = (element, options = {}) => baseMount(element, _objectSpread({
  wrappingComponent: Wrappers
}, options));

/**
 * shallow() from Enzyme but with required app-wide context providers included by default.
 */
const shallowConnected = (element, options = {}) => baseShallow(element, _objectSpread({
  wrappingComponent: Wrappers
}, options));

// Re-export everything from Enzyme
export * from 'enzyme';
// Export wrapped methods
export { mountConnected, shallowConnected };
//# sourceMappingURL=enzyme.js.map