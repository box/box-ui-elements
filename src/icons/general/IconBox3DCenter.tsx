import * as React from 'react';

import AccessibleSVG from '../accessible-svg';

import { Icon } from '../iconTypes';

const IconBox3DCenter = ({ className = '', height = 18, color = '#444', title, width = 18 }: Icon) => (
    <AccessibleSVG
        className={`box3d-center-icon ${className}`}
        height={height}
        title={title}
        viewBox="9 9 16 16"
        width={width}
    >
        <path
            className="fill-color"
            d="M20.5 13.9L16.9 12c-.5-.3-1.3-.3-1.8 0l-3.6 1.9c-.6.3-1 1-1 1.6v4.9c0 .7.4 1.3 1 1.6l3.6 1.9c.3.1.6.2.9.2.3 0 .6-.1.9-.2l3.6-1.9c.6-.3 1-1 1-1.6v-4.9c0-.6-.4-1.3-1-1.6zm-8.6 7.3c-.2-.1-.4-.5-.4-.7v-4.9c0-.3.2-.6.4-.7l3.6-1.9c.2-.1.6-.1.9 0l3.6 1.9c.1 0 .1.1.2.2L16 17v6.2c-.2 0-.3 0-.4-.1l-3.7-1.9z"
            fill={color}
        />
    </AccessibleSVG>
);

export default IconBox3DCenter;
