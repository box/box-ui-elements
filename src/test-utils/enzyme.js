/* eslint-disable import/no-extraneous-dependencies */
import * as React from 'react';
import { mount as baseMount, shallow as baseShallow } from 'enzyme';

// Global providers that need setup
import { ThemeProvider } from 'styled-components';
import defaultTheme from '../styles/theme';

// Wrap this around every component so they can use app-wide context.
// Prevents breaking due to missing providers.
// See https://github.com/airbnb/enzyme/blob/master/docs/api/ReactWrapper/getWrappingComponent.md
/* eslint-disable react/prop-types */
const Wrappers = ({ children }) => {
    return <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>;
};

/**
 * mount() from Enzyme but with required app-wide context providers included by default
 */
const mountConnected = (element, options = {}) => baseMount(element, { wrappingComponent: Wrappers, ...options });

/**
 * shallow() from Enzyme but with required app-wide context providers included by default.
 */
const shallowConnected = (element, options = {}) => baseShallow(element, { wrappingComponent: Wrappers, ...options });

// Re-export everything from Enzyme
export * from 'enzyme';
// Export wrapped methods
export { mountConnected, shallowConnected };
