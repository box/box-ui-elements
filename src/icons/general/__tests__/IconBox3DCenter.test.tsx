import React from 'react';
import { shallow } from 'enzyme';

import IconBox3DCenter from '../IconBox3DCenter';

describe('icons/general/IconBox3DCenter', () => {
    const getWrapper = (props = {}) => shallow(<IconBox3DCenter {...props} />);

    it('should correctly render icon with default values', () => {
        const wrapper = getWrapper();

        expect(wrapper).toMatchSnapshot();
    });

    it('should correctly render icon with all props specified', () => {
        const wrapper = getWrapper({
            className: 'test',
            color: '#987654',
            height: 100,
            title: 'title',
            width: 200,
        });

        expect(wrapper).toMatchSnapshot();
    });
});
