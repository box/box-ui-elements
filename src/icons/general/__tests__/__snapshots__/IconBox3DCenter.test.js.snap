// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icons/general/IconBox3DCenter should correctly render icon with all props specified 1`] = `
<AccessibleSVG
  className="box3d-center-icon test"
  height={100}
  title="title"
  viewBox="9 9 16 16"
  width={200}
>
  <path
    className="fill-color"
    d="M20.5 13.9L16.9 12c-.5-.3-1.3-.3-1.8 0l-3.6 1.9c-.6.3-1 1-1 1.6v4.9c0 .7.4 1.3 1 1.6l3.6 1.9c.3.1.6.2.9.2.3 0 .6-.1.9-.2l3.6-1.9c.6-.3 1-1 1-1.6v-4.9c0-.6-.4-1.3-1-1.6zm-8.6 7.3c-.2-.1-.4-.5-.4-.7v-4.9c0-.3.2-.6.4-.7l3.6-1.9c.2-.1.6-.1.9 0l3.6 1.9c.1 0 .1.1.2.2L16 17v6.2c-.2 0-.3 0-.4-.1l-3.7-1.9z"
    fill="#987654"
  />
</AccessibleSVG>
`;

exports[`icons/general/IconBox3DCenter should correctly render icon with default values 1`] = `
<AccessibleSVG
  className="box3d-center-icon "
  height={18}
  viewBox="9 9 16 16"
  width={18}
>
  <path
    className="fill-color"
    d="M20.5 13.9L16.9 12c-.5-.3-1.3-.3-1.8 0l-3.6 1.9c-.6.3-1 1-1 1.6v4.9c0 .7.4 1.3 1 1.6l3.6 1.9c.3.1.6.2.9.2.3 0 .6-.1.9-.2l3.6-1.9c.6-.3 1-1 1-1.6v-4.9c0-.6-.4-1.3-1-1.6zm-8.6 7.3c-.2-.1-.4-.5-.4-.7v-4.9c0-.3.2-.6.4-.7l3.6-1.9c.2-.1.6-.1.9 0l3.6 1.9c.1 0 .1.1.2.2L16 17v6.2c-.2 0-.3 0-.4-.1l-3.7-1.9z"
    fill="#444"
  />
</AccessibleSVG>
`;
